pandas/util/testing.py

```diff
@@ -1419,6 +1419,31 @@ def assert_copy(iter1, iter2, **eql_kwargs):
                                     % (type(elem1), type(elem2)))
 
 
+def assert_numpy_arrays_copy(iter1, iter2):
+    """
+    iter1, iter2: iterables that produce numpy objects
+
+    Checks that the elements are equal, but stored in different memory areas
+    """
```

Comment:
though *you could* simply add a argument to ``assert_numpy_array_equal`` something like

``check_same=None|'copy'|'same'`` (which does this)